internal/spoke_k8s_client/spoke_k8s_client_manager.go

```diff
@@ -0,0 +1,44 @@
+package spoke_k8s_client
+
+import (
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// SpokeK8sClientManager maintains a set of spoke clients
+
+//go:generate mockgen --build_flags=--mod=mod -package=spoke_k8s_client -destination=mock_spoke_k8s_client_manager.go . SpokeK8sClientManager
+type SpokeK8sClientManager interface {
+	AddFromSecret(secret *corev1.Secret) (SpokeK8sClient, error)
+}
+
+type spokeK8sClientManager struct {
```

Comment:
What happens when a kubeconfig changes? We need a way to handle that. Storing a hash of the kubeconfig contents is one way that could be identified, so the old client could be evicted from the cache.

Reply:
Yeah good point, added kubeconfig hash to the cache map.